source/docs/software/actuators/pneumatics.rst

```diff
@@ -115,6 +115,47 @@ Double solenoids are controlled by the DoubleSolenoid class in WPILib. These are
         exampleDouble.Set(frc::DoubleSolenoid::Value::kForward);
         exampleDouble.Set(frc::DoubleSolenoid::Value::kReverse);
 
+Toggling Solenoids
+------------------
+
+Solenoids can be switched from one output to the other (known as toggling) by using the `.toggle()` method.
+
+.. note::
+   Since a DoubleSolenoid defaults to off you will have to set it before it can be toggled.
+
+.. tabs::
+
+   .. code-tab:: java
+
+      Solenoid exampleSingle = new Solenoid(0);
+      DoubleSolenoid exampleDouble = new DoubleSolenoid(1, 2);
+
+      // Initialize the DoubleSolenoid so it knows where to start.  Not required for single solenoids.
+      exampleDouble.set(kReverse);
+
+      YButton.whenPressed(m_intake::toggleSolenoids);
+
+      // In subsystem
```

Comment:
I'd get rid of the subsystem comment. Remember that it's easier to go from non-commandbased -> commandbased but the other way around is harder.

Reply:
I was actively looking for feedback on this in discord and the agreement was this was fine, but I can change it.

so you want:
if(YButton) {
  toggle
  toggle
}

https://docs.wpilib.org/en/stable/docs/software/wpilib-tools/shuffleboard/advanced-usage/shuffleboard-tuning-pid.html?highlight=whenPressed

Reply:
take a look at the change I made